### DISCUSSION TITLE
loosely-coupled-events-in-net-and-harvesting-open-source-projects/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#43] [Blog comments]
# loosely-coupled-events-in-net-and-harvesting-open-source-projects/

https://blogs.taiga.nl/martijn/posts/loosely-coupled-events-in-net-and-harvesting-open-source-projects/

**Comment by disqus-comments-migrator** [comment]
<em>mick commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 28 2011, 09:52</em>

---

<p>i'd look at leveraging the container more when your building up the list of IEventListeners. ask the container to resolve out any class that implements the specific interface IEventListener<br>This way your only resolving what you need each time. <br>You'll have to change your publisher but it would be more performant.</p>

**Comment by disqus-comments-migrator** [comment]
<em>mick commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 28 2011, 09:53</em>

---

<p>that was resolve a list of handlers for the specific interface <code>IEventListener[MyEvent]  </code></p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 28 2011, 11:08</em>

---

<p>Sure, that would make things probably more performant, but how would you do that without referencing the container itself? Inject a factory?</p>
